node-it/src/test/scala/com/wavesplatform/it/asset/InvokeReissueBurnAssetSuite.scala

```diff
@@ -0,0 +1,321 @@
+      val acc = createDapp(script(data))
+
+      val fee = invocationCost(if (isNft(data)) 0 else 1)
+      val tx  = issue(acc, method, data, fee)
+
+      validateIssuedAssets(acc, tx, data, method = method)
+      sender.balanceDetails(acc).regular shouldBe (initialWavesBalance - setScriptPrice - fee)
+    }
+
+    for (data <- Seq(simpleNonreissuableAsset, simpleReissuableAsset)) s"${data("type")} asset could be partially burned" in {
+      val acc            = createDapp(script(data))
+      val fee            = invocationCost(if (isNft(data)) 0 else 1)
+      val txIssue        = issue(acc, method, data, fee)
+      val assetId        = validateIssuedAssets(acc, txIssue, data, method = method)
+      val burnQuantity   = 1000
+      val remainQuantity = data("quantity").asInstanceOf[Number].longValue - burnQuantity
+
+      burn(acc, method, assetId, burnQuantity)
+
+      sender.assetsDetails(assetId).quantity shouldBe remainQuantity
+      sender.assertAssetBalance(acc, assetId, remainQuantity)
+    }
+
+    for (data <- Seq(simpleNonreissuableAsset, simpleReissuableAsset, nftAsset)) s"${data("type")} could be fully burned" in {
+      val acc     = createDapp(script(data))
+      val fee     = invocationCost(if (isNft(data)) 0 else 1)
+      val txIssue = invokeScript(acc, "issueAsset", fee = fee)
+      val assetId = validateIssuedAssets(acc, txIssue, data, method = method)
+
+      burn(acc, method, assetId, data("quantity").asInstanceOf[Number].intValue)
+
+      sender.assetsDetails(assetId).quantity shouldBe 0
+      sender.assertAssetBalance(acc, assetId, 0)
+    }
+
+    "Reissuable asset could be reissued" in {
+      val acc               = createDapp(script(simpleReissuableAsset))
+      val txIssue           = invokeScript(acc, "issueAsset")
+      val assetId           = validateIssuedAssets(acc, txIssue, simpleReissuableAsset, method = method)
+      val initialQuantity   = simpleReissuableAsset("quantity").asInstanceOf[Number].longValue
+      val addedQuantity     = 100500
+      val initialReissuable = simpleReissuableAsset("isReissuable").asInstanceOf[Boolean].booleanValue
+
+      reissue(acc, method, assetId, addedQuantity, !initialReissuable)
+
+      sender.assetsDetails(assetId).reissuable shouldBe !initialReissuable
+      sender.assetsDetails(assetId).quantity shouldBe initialQuantity + addedQuantity
+      sender.assertAssetBalance(acc, assetId, initialQuantity + addedQuantity)
+    }
+
+    "Non-reissuable asset could not be reissued" in {
+      val acc     = createDapp(script(simpleNonreissuableAsset))
+      val txIssue = invokeScript(acc, "issueAsset")
+      val assetId = validateIssuedAssets(acc, txIssue, simpleNonreissuableAsset, method = method)
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, 100500, false),
+        "State check failed. Reason: Asset is not reissuable"
+      )
+    }
+  }
+
+  "Restrictions in @Callable" - {
```

Comment:
What if two assets are issued with the same nonce

Reply:
ok